src/pocketmine/Player.php

```diff
@@ -3321,7 +3322,9 @@ public function handleDataPacket(DataPacket $packet){
 
 				$tile = $this->level->getTile($this->temporalVector->setComponents($packet->x, $packet->y, $packet->z));
 				if($tile instanceof ItemFrame){
-					if($this->isSpectator()){
+					$this->server->getPluginManager()->callEvent($ev = new ItemFrameDropItemEvent($this, $tile->getBlock(), $tile, $tile->getItem()));
+
+					if($this->isSpectator() or $ev->isCancelled()){
```

Comment:
Maybe better to just setCancelled() if the player is spectator instead of hardcoding this. Then plugins can override it.